lib/cli/cli.js

```diff
@@ -84,13 +84,55 @@ class CLI {
 
   /**
    * @private
-   * @method run
+   * @method setEnvironment
    * @param environment
+   */
+  setEnvironment(environment) {
```

Comment:
If we cant pass this in via the constructor, i think we should restore the original functionality and provide via `run`, and simply have run update an internal/private `._environment` property, that `maybeMakeCommand` can use.

Also, then we should make `maybeMakeCommand` fail, if the `this._environment` is not set.